src/monaca.js

```diff
@@ -132,6 +162,106 @@
     this._monacaData = this._loadAllData();
   };
 
+
+  Monaca.prototype._generateUUIDv4 = function(a, b) {
+    for (
+      b = a = ''; a++ < 36; b += a * 51 & 52 ?
+      (
+        a ^ 15 ?
+        8 ^ Math.random() *
+        (a ^ 20 ? 16 : 4) :
+        4
+      ).toString(16) :
+      '-'
+    );
+    return b;
+  };
+
+  Monaca.prototype.getTrackId = function() {
+    if (this.getData('trackId')) {
+      return Q.resolve(this.getData('trackId'));
+    }
+    return this.setData({
+      trackId: this._generateUUIDv4()
+    }).then(Q.resolve.bind(null, this.getData('trackId')));
+  };
+
+  /**
+   * @method
+   * @memberof Monaca
+   * @description
+   *  Reports any event to Monaca backend. It keeps the previous promise flow.
+   * @param {object} report - Report object. Must contain 'event'.
+   *  Optional values are 'arg1' and 'otherArgs'.
+   * @param {any} resolvedValue - Optional. This parameter will be returned by the promise.
+   * @return {Promise}
+   * @example
+   *   monaca.reportAnalytics({event: 'create'})
+   *     .then(nextMethod)
+   */
+  Monaca.prototype.reportAnalytics = function(report, resolvedValue) {
+    return this.getTrackId().then(
+      function(trackId) {
+        var form = extend({}, report, { event: 'monaca-local-' + report.event }, {
```

Comment:
Can we change the event name to `monaca-lib-` instead of `monaca-local-`? That looks more sense to me.